src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs

```diff
@@ -1146,6 +1245,58 @@ private unsafe IReadOnlyList<RawPointerPoint> CreateIntermediatePoints(MOUSEMOVE
 
         }
 
+        private IReadOnlyList<RawPointerPoint>? CreateIntermediatePoints(uint lastPacket, uint currentPacketSerial)
+        {
+            if (lastPacket + 1 >= currentPacketSerial)
+            {
+                return null;
+            }
+
+            s_intermediatePointsPooledList.Clear();
+            s_intermediatePointsPooledList.Capacity = (int)(currentPacketSerial - lastPacket - 1);
+
+            for (uint pkSerial = lastPacket + 1; pkSerial < currentPacketSerial; pkSerial++)
+            {
+                if (s_lastWintabPackets.TryGetValue(pkSerial, out var packet))
+                {
+                    s_intermediatePointsPooledList.Add(CreateRawPointerPoint(packet));
+                }
+            }
+
+            return s_intermediatePointsPooledList;
+        }
+
+        private void FlushIntermediatePackets(uint count)
+        {
+            uint actualSize = 0;
+            var packets = _wnData.GetDataPackets(count, true, ref actualSize);
+            foreach (var packet in packets)
+            {
+                s_lastWintabPackets[packet.pkSerialNumber] = packet;
+            }
+
+            if (s_lastWintabPackets.Count > MaxWintabPacketHistorySize)
+            {
+                var keysToRemove = s_lastWintabPackets.Keys
+                    .OrderBy(k => k)
+                    .Take(s_lastWintabPackets.Count - MaxWintabPacketHistorySize)
+                    .ToList();
+
+                foreach (var key in keysToRemove)
+                {
+                    s_lastWintabPackets.Remove(key);
+                }
+            }
+        }
```

Comment:
I know this is not ideal, but Wintab API does not have anything like Ink to return packets with starting index argument. Buffering it into local dictionary on each movement also prevents buffer overflows within Wintab itself.